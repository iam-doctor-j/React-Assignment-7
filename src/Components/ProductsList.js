import React from 'react';
import Product from './Product';

export default class ProductsList extends React.Component {
    
    render() {
        // let headings = this.props.data && Object.keys(this.props.data[0]);

        // let infoItems = headings && headings.map((item) => {
        //     return(
        //         <th key={item}>
        //             { item === 'id' ? 'ID' : item === 'productName' ? 'Product Name' : item }
        //         </th>
        //     );
        // });
        let infoItems = 
            <>
                <th>ID</th>
                <th>Product Name</th>
                <th>Quantity</th>
                <th>Price</th>
            </>
        

        let listItems = this.props.data && this.props.data.map((item, i) => {
            return(
                <Product key={i} id={item.id} name={item.productName} quantity={item.Quantity} price={item.Price} />
            );
        });

        console.log(this.props);
        return(
            <table border="1">
                <thead>
                    <tr key="head">
                        { infoItems }
                    </tr>
                </thead>
                <tbody>
                    { listItems }
                </tbody>
            </table>
        );
    }
}