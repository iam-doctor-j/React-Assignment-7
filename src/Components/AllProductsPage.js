import React from 'react';
import DataApi from '../Data/DataApi';
import ProductsList from './ProductsList';

export default class AllProductsPage extends React.Component {
    constructor() {
        super();
        this.state = {
            data: DataApi.getData()
        }
    }

    render() {
        return(
            <div style={{ marginLeft: 10 }}>
                <h1>
                    Products List
                </h1>
                <ProductsList data={ this.state.data } />
            </div>
        );
    }
}