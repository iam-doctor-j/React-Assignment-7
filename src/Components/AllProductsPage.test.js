import React from 'react';
import AllProductsPage from './AllProductsPage';
import { mount, shallow } from 'enzyme';
import ProductsList from './ProductsList';

describe('AllProductsPage', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<AllProductsPage/>);
    });

    afterEach(() => {
        wrapper.unmount(<AllProductsPage/>);
    });

    test('Snapshot renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('AllProductsPage', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<AllProductsPage/>);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('renders heading name correctly', () => {
        expect(wrapper.find('h1').render().text()).toEqual('Products List');
    });

    it('renders one ProductsList', () => {
        expect(wrapper.find(ProductsList).length).toBe(1);
    })
});