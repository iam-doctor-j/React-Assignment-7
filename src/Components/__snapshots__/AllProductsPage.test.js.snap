// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProductsPage Snapshot renders correctly 1`] = `
<AllProductsPage>
  <div
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <h1>
      Products List
    </h1>
    <ProductsList
      data={
        Array [
          Object {
            "Price": 15999,
            "Quantity": 2,
            "id": 1,
            "productName": "Nokia 6.1 Plus",
          },
          Object {
            "Price": 14999,
            "Quantity": 1,
            "id": 2,
            "productName": "Realme 6",
          },
          Object {
            "Price": 64000,
            "Quantity": 1,
            "id": 3,
            "productName": "Lenovo L340 Gaming",
          },
        ]
      }
    >
      <table
        border="1"
      >
        <thead>
          <tr
            key="head"
          >
            <th>
              ID
            </th>
            <th>
              Product Name
            </th>
            <th>
              Quantity
            </th>
            <th>
              Price
            </th>
          </tr>
        </thead>
        <tbody>
          <Product
            id={1}
            key="0"
            name="Nokia 6.1 Plus"
            price={15999}
            quantity={2}
          >
            <tr
              key="1"
            >
              <td>
                1
              </td>
              <td>
                Nokia 6.1 Plus
              </td>
              <td>
                2
              </td>
              <td>
                Rs. 
                15999
              </td>
            </tr>
          </Product>
          <Product
            id={2}
            key="1"
            name="Realme 6"
            price={14999}
            quantity={1}
          >
            <tr
              key="2"
            >
              <td>
                2
              </td>
              <td>
                Realme 6
              </td>
              <td>
                1
              </td>
              <td>
                Rs. 
                14999
              </td>
            </tr>
          </Product>
          <Product
            id={3}
            key="2"
            name="Lenovo L340 Gaming"
            price={64000}
            quantity={1}
          >
            <tr
              key="3"
            >
              <td>
                3
              </td>
              <td>
                Lenovo L340 Gaming
              </td>
              <td>
                1
              </td>
              <td>
                Rs. 
                64000
              </td>
            </tr>
          </Product>
        </tbody>
      </table>
    </ProductsList>
  </div>
</AllProductsPage>
`;
